### PR TITLE
[TF FE] Add translators for IsInf and IsNan and test it

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Add translators for `IsInf` and `IsNan` operations and the layer tests.

**Tickets:** 94752, 94720

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

### CODE REVIEW

File: tests/layer_tests/tensorflow_tests/test_tf_IsInf.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        return tf_net, None
+
+    test_data_basic = [
+        dict(x_shape=[8], x_type=tf.float32),
+        dict(x_shape=[3, 7], x_type=tf.float32),
+        dict(x_shape=[4, 3, 2], x_type=tf.float32),
+    ]
+
+    @pytest.mark.parametrize("params", test_data_basic)
```
Comment (itikhono) [inline tests/layer_tests/tensorflow_tests/test_tf_IsInf.py:37]: do we plan to enable it in precommit?

**Review by itikhono — APPROVED** [review]

File: src/frontends/tensorflow/src/op/is_inf.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+using namespace std;
+using namespace ov::opset10;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+OutputVector translate_is_inf_op(const NodeContext& node) {
```
Comment (itikhono) [inline src/frontends/tensorflow/src/op/is_inf.cpp:16]: why do we need a separate translator for this? can we use translate_unary_op function in op_table.cpp?

File: tests/layer_tests/tensorflow_tests/test_tf_IsInf.py (hunk 1 above)
Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_IsInf.py:37]: Yes, once it is implemented in CPU

File: src/frontends/tensorflow/src/op/is_inf.cpp (hunk 2 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/op/is_inf.cpp:16]: Let us do this

File: tests/layer_tests/tensorflow_tests/test_tf_IsInf.py (hunk 1 above)
Comment (itikhono) [reply tests/layer_tests/tensorflow_tests/test_tf_IsInf.py:37]: What is the current test result? Do we use a reference to calculate values?

Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_IsInf.py:37]: No, we don't. They are failing with this error:

```
>       net = core.read_model(self.model, self.weights)
E       RuntimeError: Check 'false' failed at openvino\src\frontends\common\src\frontend.cpp:54:
E       Converting input model
E       Cannot create IsNaN layer is_nan id:1 from unsupported opset: opset10
```

Comment (itikhono) [reply tests/layer_tests/tensorflow_tests/test_tf_IsInf.py:37]: Should we temporary disable these tests with xfail and create a new ticket to enable it once it is implemented in CPU?

Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_IsInf.py:37]: Added `xfail` and refer to the CPU implementation. I will put a comment there to remove this `xfail` when they post PR.

File: src/frontends/tensorflow/src/op/is_inf.cpp (hunk 2 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/op/is_inf.cpp]: Done

**Review by itikhono — APPROVED** [review]

**Pull request merged**